### PR TITLE
Update changelog for v10.0.0-beta.17

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Review by kiryldz — CHANGES_REQUESTED** [review]
@Chaoba we need to pull up and also add to changelog all the changes from gl-native beta.18 (our previous release was using just gl-native beta.17) and also common.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,38 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-beta.17 - April 12, 2021
+## Breaking changes ⚠️
+* [Annotation plugin] Rename annotation classes ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
+* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contains additional properties(source (id of the source), sourceLayer (id of the source's layer) ,state (feature's state)) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
+* [core] Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
+* [core] Remove old map#drag API and the AnimationOptions API
+* [core] Move camera related methods to CameraManager
```
Comment (kiryldz) [inline CHANGELOG.md:11]: Does not reflect to Android.

**Comment by kiryldz** [comment]
@Chaoba still need changelog from common - especially about network reachability.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,41 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-beta.17 - April 12, 2021
+## Breaking changes ⚠️
+* [Annotation plugin] Rename annotation classes ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
```
Comment (kiryldz) [inline CHANGELOG.md:7]: Please also add how exactly classes were renamed - we were already asked about it several times.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,41 @@
 
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-beta.17 - April 12, 2021
+## Breaking changes ⚠️
+* [Annotation plugin] Rename annotation classes(Symbol -> PointAnnotation, Circle -> CircleAnnotation,Line -> PolylineAnnotation,Fill -> PolygonAnnotation) ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
```
Comment (kiryldz) [inline CHANGELOG.md:7]: Missing renames about options classes. And please align them like that for better readability:
 - Symbol -> PointAnnotation
 -  ...

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,41 @@
...
+* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contains additional properties(source (id of the source), sourceLayer (id of the source's layer) ,state (feature's state)) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
```
Comment (kiryldz) [inline CHANGELOG.md:8]:
```suggestion
* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
```

File: CHANGELOG.md (hunk 3 above)
Comment (Chaoba) [reply CHANGELOG.md:7]: IMHO it will be too verbose to mention all the details in changelog. For breaking changes, we will add them to the migration guide

Comment (kiryldz) [reply CHANGELOG.md:7]: Migration guide is for v9 -> v10 migration while those changes affect v10 only.

Comment (kiryldz) [reply CHANGELOG.md:7]: In general we have section in migration guide about annotations so we must update there as well.
But I would also provide that information here - @pengdev what are your thoughts here?

Comment (pengdev) [reply CHANGELOG.md:7]: Was it different in v9? if no I think we should state this clearly in the migration guide.
And we could also include a link to the migration guide in this entry when the migration guide is updated?

Comment (Chaoba) [reply CHANGELOG.md:7]: It's different in v9

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,41 @@
...
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-beta.17 - April 13, 2021
+## Breaking changes ⚠️
+* [Annotation plugin] Rename annotation classes, the rules are as follows and applied for Annotations/Options/Managers ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
+    - Symbol -> PointAnnotation
+    - Circle -> CircleAnnotation
+    - Line -> PolylineAnnotation
+    - Fill -> PolygonAnnotation
+* [deps] mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
```
Comment (pengdev) [inline CHANGELOG.md:12]: we can remove the [deps] here

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,46 @@
...
 Mapbox welcomes participation and contributions from everyone.
 
+# 10.0.0-beta.17 - April 14, 2021
+## Breaking changes ⚠️
+* [Annotation plugin] Rename annotation classes, the rules are as follows and applied for Annotations/Options/Managers ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
+    - Symbol -> PointAnnotation
+    - Circle -> CircleAnnotation
+    - Line -> PolylineAnnotation
+    - Fill -> PolygonAnnotation
+* mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties: source (id of the source), sourceLayer (id of the source's layer), state (feature's state) ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247))
```
Comment (knov) [inline CHANGELOG.md:12]:
```suggestion
* mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247)): 
    - source (id of the source)
    - sourceLayer (id of the source's layer)
    - state (feature's state) 
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,46 @@
...
+* Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
+* Remove old map#drag API and the AnimationOptions API
+* Don't emit MapIdle event when there is gesture and / or user animation in progress
+* Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
```
Comment (knov) [inline CHANGELOG.md:16]:
```suggestion
* Mark overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,46 @@
...
+
+## Features ✨ and improvements 🏁
+* Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.
```
Comment (knov) [inline CHANGELOG.md:19]:
```suggestion
* Integrate map engine with NetworkConnectivity API. Map will respect `setMapboxStackConnected` setting and decide whether http requests can be made.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,6 +2,46 @@
...
+* Fix distance expression parsing for geometries containing over 20k points
+* Fixed holes in the ground for terrain with negative elevation
+* Make StyleImageMissing callback a no-op after ImageManager destruction
+* Reset unprocessed event queue for destructed renderer
+* Fix clipping of fill-extrusions on near plane
+* Set 'clusterMaxZoom' to be 'maxZoom-1' when it is not provided
+* Fix crash for the case when MapSnapshotter object is destructed from within user provided callback
+* Terrain render cache now disabled during property transitions
+* Tile store: handle sparse tilesets and empty tile packs
+* Tile store: handle immutable tile packs
```
Comment (knov) [inline CHANGELOG.md:37]:
```suggestion
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -2,6 +2,47 @@
...
+    - sourceLayer (id of the source's layer)
+    - state (feature's state) 
+* Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
+* Remove old map#drag API and the AnimationOptions API
+* Don't emit MapIdle event when there is gesture and / or user animation in progress
+* Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
+
+## Features ✨ and improvements 🏁
+* Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.
+* Introducing NetworkConnectivity (offline switch).
```
Comment (pengdev) [inline CHANGELOG.md:23]:
```suggestion
* Introducing NetworkConnectivity API (offline switch). If setting setMapboxStackConnected(false), the Reachability API will report NotReachable, http requests are being blocked and if running, cancelled.
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -2,6 +2,47 @@
...
+* Adding NetworkStatus ReachableViaEthernet.
```
Comment (pengdev) [inline CHANGELOG.md:24]:
```suggestion
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -2,6 +2,47 @@
...
+* Added new CameraManager.cameraForCoordinates overload
+* Query rendered features for Circle Layer on Terrain
+* Reduce binary size of native map shared library
+
+## Bug fixes 🐞
+* Fix runtime crash if logo / attribution not enabled ([#240](https://github.com/mapbox/mapbox-maps-android/pull/240))
```
Comment (pengdev) [inline CHANGELOG.md:30]: Fixed a bug that causes map not loading when connected through ethernet.

File: CHANGELOG.md [hunk 13]
```diff
@@ -2,6 +2,47 @@
...
+* [Annotation plugin] Rename annotation classes, the rules are as follows and applied for Annotations/Options/Managers ([#227](https://github.com/mapbox/mapbox-maps-android/pull/227))
+    - Symbol -> PointAnnotation
+    - Circle -> CircleAnnotation
+    - Line -> PolylineAnnotation
+    - Fill -> PolygonAnnotation
+* mapboxMap.queryRenderedFeatures will return a new data class QueriedFeature which will contain additional properties ([#247](https://github.com/mapbox/mapbox-maps-android/pull/247)): 
+    - source (id of the source)
+    - sourceLayer (id of the source's layer)
+    - state (feature's state) 
+* Deprecate MapObserver, MapSnapshotterObserver interfaces and align sync method naming, Map#isMapLoaded, StyleManager#isStyleLoaded
```
Comment (pengdev) [inline CHANGELOG.md:16]:
```suggestion
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -2,6 +2,47 @@
...
+* Remove old map#drag API and the AnimationOptions API
```
Comment (pengdev) [inline CHANGELOG.md:17]: @kiryldz thoughts on this entry? Is this API public-facing?

File: CHANGELOG.md [hunk 15]
```diff
@@ -2,6 +2,47 @@
...
+* Don't emit MapIdle event when there is gesture and / or user animation in progress
+* Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
+
+## Features ✨ and improvements 🏁
+* Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.
+* Introducing NetworkConnectivity (offline switch).
+* Adding NetworkStatus ReachableViaEthernet.
+* Added new CameraManager.cameraForCoordinates overload
+* Query rendered features for Circle Layer on Terrain
+* Reduce binary size of native map shared library
```
Comment (pengdev) [inline CHANGELOG.md:27]:
```suggestion
* Enable identical code folding and -lto-O3 optimizations to reduce binary size of native map shared library
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -2,6 +2,47 @@
...
+* Remove old map#drag API and the AnimationOptions API
+* Don't emit MapIdle event when there is gesture and / or user animation in progress
+* Make overlay plugin experimental ([#233](https://github.com/mapbox/mapbox-maps-android/pull/233))
+
+## Features ✨ and improvements 🏁
+* Integrate map engine with NetworkConnectivity API. Map will respect setMapboxStackConnected setting and decide whether http requests can be made.
+* Introducing NetworkConnectivity (offline switch).
+* Adding NetworkStatus ReachableViaEthernet.
+* Added new CameraManager.cameraForCoordinates overload
+* Query rendered features for Circle Layer on Terrain
```
Comment (pengdev) [inline CHANGELOG.md:26]:
```suggestion
* Added support for query rendered features for Circle Layer on Terrain
```

File: CHANGELOG.md (hunk 14 above)
Comment (kiryldz) [reply CHANGELOG.md:17]: Yes, it is public, so let's leave it.

**Review by kiryldz — APPROVED** [review]

**Pull request merged**